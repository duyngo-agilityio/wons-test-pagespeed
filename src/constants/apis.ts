export const API_BASE_URL = process.env.NEXT_API_BASE_URL || '';

export const API_PATH = {
  SIGN_UP: '/auth/local/register',
  USERS: '/users',
};
