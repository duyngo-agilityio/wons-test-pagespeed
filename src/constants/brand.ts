export const BRANDS = [
  { key: 'apple', label: 'Apple' },
  { key: 'samsung', label: 'Samsung' },
  { key: 'huawei', label: 'Huawei' },
  { key: 'xioami', label: 'Xioami' },
  { key: 'oppo', label: 'Oppo' },
  { key: 'google', label: 'Google' },
];
