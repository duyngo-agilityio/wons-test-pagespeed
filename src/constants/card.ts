export enum LevelCard {
  LOW = 'low',
  MEDIUM = 'medium',
  HIGH = 'high',
}
