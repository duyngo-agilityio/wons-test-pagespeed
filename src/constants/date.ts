export const DEFAULT_RANGE_VALUE_PICKER = 7;
