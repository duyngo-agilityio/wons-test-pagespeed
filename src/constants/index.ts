export * from './apis';
export * from './auth';
export * from './date';
export * from './fonts';
export * from './images';
export * from './messages';
export * from './params';
export * from './regex';
export * from './routes';
export * from './sidebar';
export * from './themes';
export * from './typography';
export * from './messages';
export * from './regex';
export * from './apis';
export * from './auth';
export * from './invoices';
export * from './select';
export * from './pagination';
