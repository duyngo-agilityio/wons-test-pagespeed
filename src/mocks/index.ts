export * from './customer';
export * from './recentServices';
export * from './tabs';
