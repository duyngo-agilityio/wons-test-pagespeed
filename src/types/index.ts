export * from './auth';
export * from './date';
export * from './params';
export * from './select';
export * from './strapi';
export * from './table';
export * from './toast';
export * from './invoice';
export * from './customer';
