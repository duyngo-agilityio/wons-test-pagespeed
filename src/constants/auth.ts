// Types
import { TRole } from '@/types';

export const AUTH_METHODS = {
  CREDENTIALS: 'credentials',
};

export const ERROR_TYPES = {
  CREDENTIALS_SIGN_IN: 'CallbackRouteError',
};

export const ROLES: TRole[] = [
  {
    id: 3,
    name: 'Admin',
  },
  {
    id: 4,
    name: 'User',
  },
];
