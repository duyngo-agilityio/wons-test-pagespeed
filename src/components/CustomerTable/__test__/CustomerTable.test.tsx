import { render } from '@testing-library/react';
import { act } from 'react';

// types
import { TCustomerDataResponse } from '@/types';

// mocks
import { CUSTOMER_MOCK } from '@/mocks';

// components
import CustomersTable from '../index';

// Constants
import { DEFAULT_PAGE } from '@/constants';

const mockCustomersData: TCustomerDataResponse[] = [
  {
    id: 1,
    attributes: CUSTOMER_MOCK[0],
  },
  {
    id: 2,
    attributes: CUSTOMER_MOCK[1],
  },
];

describe('CustomersTable Component', () => {
  const mockOnEdit = jest.fn();
  const mockOnDelete = jest.fn();

  it('renders the table headers and customer data correctly', async () => {
    const { container } = render(
      <CustomersTable
        onSort={jest.fn()}
        isReadOnly={false}
        data={mockCustomersData}
        onEdit={mockOnEdit}
        onDelete={mockOnDelete}
        pageCount={DEFAULT_PAGE}
      />,
    );

    await act(async () => {
      expect(container).toMatchSnapshot();
    });
  });

  it('renders correctly with undefined attributes', async () => {
    const { getAllByText } = render(
      <CustomersTable
        onSort={jest.fn()}
        data={undefined as unknown as TCustomerDataResponse[]}
        onEdit={mockOnEdit}
        onDelete={mockOnDelete}
        pageCount={DEFAULT_PAGE}
      />,
    );

    await act(async () => {
      expect(getAllByText('')).toHaveLength(31);
    });
  });

  it('renders correctly with empty attributes', async () => {
    const { getAllByText } = render(
      <CustomersTable
        onSort={jest.fn()}
        data={[]}
        onEdit={mockOnEdit}
        onDelete={mockOnDelete}
        pageCount={DEFAULT_PAGE}
      />,
    );

    await act(async () => {
      expect(getAllByText('')[1]).toBeInTheDocument();
    });
  });
});
