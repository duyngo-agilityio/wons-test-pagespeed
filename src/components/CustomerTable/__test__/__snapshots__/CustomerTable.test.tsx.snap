// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomersTable Component renders the table headers and customer data correctly 1`] = `
<div>
  <div
    class="flex flex-col gap-10"
  >
    <div
      class="flex-col relative gap-4 w-full grid"
    >
      <div
        class="z-0 flex flex-col relative justify-between gap-4 bg-content1 overflow-auto rounded-large w-full p-0 shadow-none dark:bg-gray-400"
      >
        <table
          aria-describedby=""
          class="min-w-full h-auto table-auto w-full border-separate border-spacing-x-0 border-spacing-y-[10px] bg-gray-50 dark:bg-gray-600"
          id="react-aria-:r0:"
          role="grid"
          tabindex="0"
        >
          <thead
            class="[&>tr]:first:rounded-lg"
            role="rowgroup"
          >
            <tr
              class="group outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2"
              role="row"
            >
              <th
                class="group h-10 align-middle whitespace-nowrap text-foreground-500 text-tiny first:rounded-s-lg last:rounded-e-lg data-[sortable=true]:cursor-pointer data-[hover=true]:text-foreground-400 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-start first:rounded-l-none last:rounded-r-none py-0 cursor-pointer font-normal bg-gray-50 dark:bg-gray-600 px-[17px]"
                data-key="Name0"
                data-testid="sort-btn"
                id="react-aria-:r0:-Name0"
                role="columnheader"
                tabindex="-1"
              >
                <div
                  class="flex items-center gap-2"
                >
                  Name
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="m98 190.06 139.78 163.12a24 24 0 0 0 36.44 0L414 190.06c13.34-15.57 2.28-39.62-18.22-39.62h-279.6c-20.5 0-31.56 24.05-18.18 39.62z"
                    />
                  </svg>
                </div>
              </th>
              <th
                class="group h-10 align-middle whitespace-nowrap text-foreground-500 text-tiny first:rounded-s-lg last:rounded-e-lg data-[sortable=true]:cursor-pointer data-[hover=true]:text-foreground-400 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-start first:rounded-l-none last:rounded-r-none py-0 cursor-pointer font-normal bg-gray-50 dark:bg-gray-600 px-[17px]"
                data-key="Email1"
                data-testid="sort-btn"
                id="react-aria-:r0:-Email1"
                role="columnheader"
                tabindex="-1"
              >
                <div
                  class="flex items-center gap-2"
                >
                  Email
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="m98 190.06 139.78 163.12a24 24 0 0 0 36.44 0L414 190.06c13.34-15.57 2.28-39.62-18.22-39.62h-279.6c-20.5 0-31.56 24.05-18.18 39.62z"
                    />
                  </svg>
                </div>
              </th>
              <th
                class="group h-10 align-middle whitespace-nowrap text-foreground-500 text-tiny first:rounded-s-lg last:rounded-e-lg data-[sortable=true]:cursor-pointer data-[hover=true]:text-foreground-400 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-start first:rounded-l-none last:rounded-r-none py-0 cursor-pointer font-normal bg-gray-50 dark:bg-gray-600 px-[17px]"
                data-key="Phone Number2"
                data-testid="sort-btn"
                id="react-aria-:r0:-PhoneNumber2"
                role="columnheader"
                tabindex="-1"
              >
                <div
                  class="flex items-center gap-2"
                >
                  Phone Number
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="m98 190.06 139.78 163.12a24 24 0 0 0 36.44 0L414 190.06c13.34-15.57 2.28-39.62-18.22-39.62h-279.6c-20.5 0-31.56 24.05-18.18 39.62z"
                    />
                  </svg>
                </div>
              </th>
              <th
                class="group h-10 align-middle whitespace-nowrap text-foreground-500 text-tiny first:rounded-s-lg last:rounded-e-lg data-[sortable=true]:cursor-pointer data-[hover=true]:text-foreground-400 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-start first:rounded-l-none last:rounded-r-none py-0 cursor-pointer font-normal bg-gray-50 dark:bg-gray-600 px-[17px]"
                data-key="Gender3"
                data-testid="sort-btn"
                id="react-aria-:r0:-Gender3"
                role="columnheader"
                tabindex="-1"
              >
                <div
                  class="flex items-center gap-2"
                >
                  Gender
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="m98 190.06 139.78 163.12a24 24 0 0 0 36.44 0L414 190.06c13.34-15.57 2.28-39.62-18.22-39.62h-279.6c-20.5 0-31.56 24.05-18.18 39.62z"
                    />
                  </svg>
                </div>
              </th>
              <th
                class="group h-10 align-middle whitespace-nowrap text-foreground-500 text-tiny first:rounded-s-lg last:rounded-e-lg data-[sortable=true]:cursor-pointer data-[hover=true]:text-foreground-400 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-start first:rounded-l-none last:rounded-r-none py-0 cursor-pointer font-normal bg-gray-50 dark:bg-gray-600 px-[17px]"
                data-key="undefined4"
                data-testid="sort-btn"
                id="react-aria-:r0:-undefined4"
                role="columnheader"
                tabindex="-1"
              >
                <div
                  class="flex items-center gap-2"
                />
              </th>
            </tr>
            <tr
              aria-hidden="true"
              class="w-px h-px block"
              style="margin-left: 0.25rem; margin-top: 0.25rem;"
              tabindex="-1"
            />
          </thead>
          <tbody
            role="rowgroup"
          >
            <tr
              aria-labelledby="react-aria-:r0:-1-Name0"
              class="group outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 bg-white dark:bg-gray-400"
              data-first="true"
              data-id="1"
              data-key="1"
              role="row"
              tabindex="-1"
            >
              <td
                class="relative align-middle whitespace-normal font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start first:rounded-l-lg last:rounded-r-lg text-[10px] sm:text-[14px] leading-[18px] py-[17px] px-[17px]"
                data-key="table-cell-0"
                id="react-aria-:r0:-1-Name0"
                role="rowheader"
                tabindex="-1"
              >
                <div
                  class="flex gap-3.5 items-center"
                >
                  <div
                    class="relative w-9 h-9 rounded-full"
                  >
                    <img
                      alt="customer avatar"
                      class="rounded-full "
                      data-nimg="fill"
                      decoding="async"
                      loading="lazy"
                      sizes="100vw"
                      src="http://localhost/_next/image?url=https%3A%2F%2Fimg-cdn.pixlr.com%2Fimage-generator%2Fhistory%2F65bb506dcb310754719cf81f%2Fede935de-1138-4f66-8ed7-44bd16efc709%2Fmedium.webp&w=3840&q=75"
                      srcset="/_next/image?url=https%3A%2F%2Fimg-cdn.pixlr.com%2Fimage-generator%2Fhistory%2F65bb506dcb310754719cf81f%2Fede935de-1138-4f66-8ed7-44bd16efc709%2Fmedium.webp&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fimg-cdn.pixlr.com%2Fimage-generator%2Fhistory%2F65bb506dcb310754719cf81f%2Fede935de-1138-4f66-8ed7-44bd16efc709%2Fmedium.webp&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fimg-cdn.pixlr.com%2Fimage-generator%2Fhistory%2F65bb506dcb310754719cf81f%2Fede935de-1138-4f66-8ed7-44bd16efc709%2Fmedium.webp&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fimg-cdn.pixlr.com%2Fimage-generator%2Fhistory%2F65bb506dcb310754719cf81f%2Fede935de-1138-4f66-8ed7-44bd16efc709%2Fmedium.webp&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fimg-cdn.pixlr.com%2Fimage-generator%2Fhistory%2F65bb506dcb310754719cf81f%2Fede935de-1138-4f66-8ed7-44bd16efc709%2Fmedium.webp&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fimg-cdn.pixlr.com%2Fimage-generator%2Fhistory%2F65bb506dcb310754719cf81f%2Fede935de-1138-4f66-8ed7-44bd16efc709%2Fmedium.webp&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fimg-cdn.pixlr.com%2Fimage-generator%2Fhistory%2F65bb506dcb310754719cf81f%2Fede935de-1138-4f66-8ed7-44bd16efc709%2Fmedium.webp&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fimg-cdn.pixlr.com%2Fimage-generator%2Fhistory%2F65bb506dcb310754719cf81f%2Fede935de-1138-4f66-8ed7-44bd16efc709%2Fmedium.webp&w=3840&q=75 3840w"
                      style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
                    />
                  </div>
                  <p
                    class="font-dm-sans text-blue-800 dark:text-white text-md text-nowrap"
                  >
                    John Doe
                  </p>
                </div>
              </td>
              <td
                class="relative align-middle whitespace-normal font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start first:rounded-l-lg last:rounded-r-lg text-[10px] sm:text-[14px] leading-[18px] py-[17px] px-[17px]"
                data-key="table-cell-1"
                role="gridcell"
                tabindex="-1"
              >
                <div
                  class="flex gap-2.5 items-center"
                >
                  <p
                    class="font-dm-sans text-blue-800 dark:text-white text-md text-nowrap"
                  >
                    john.doe@example.com
                  </p>
                </div>
              </td>
              <td
                class="relative align-middle whitespace-normal font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start first:rounded-l-lg last:rounded-r-lg text-[10px] sm:text-[14px] leading-[18px] py-[17px] px-[17px]"
                data-key="table-cell-2"
                role="gridcell"
                tabindex="-1"
              >
                <p
                  class="font-dm-sans text-blue-800 dark:text-white text-md text-nowrap"
                >
                  123-456-7890
                </p>
              </td>
              <td
                class="relative align-middle whitespace-normal font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start first:rounded-l-lg last:rounded-r-lg text-[10px] sm:text-[14px] leading-[18px] py-[17px] px-[17px]"
                data-key="table-cell-3"
                role="gridcell"
                tabindex="-1"
              >
                <div
                  class="relative max-w-fit min-w-min inline-flex items-center justify-between box-border whitespace-nowrap h-7 text-small text-default-foreground w-[100px] px-2 rounded-[20px] bg-teal-500/10 dark:bg-green-200"
                >
                  <span
                    class="flex-1 font-normal px-2 text-center text-md capitalize text-teal-500 dark:text-teal-300"
                  >
                    male
                  </span>
                </div>
              </td>
              <td
                class="relative align-middle whitespace-normal font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start first:rounded-l-lg last:rounded-r-lg text-[10px] sm:text-[14px] leading-[18px] py-[17px] px-[17px]"
                data-key="table-cell-4"
                role="gridcell"
                tabindex="-1"
              >
                <button
                  aria-expanded="false"
                  aria-haspopup="true"
                  class="group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none text-default-foreground data-[hover=true]:opacity-hover z-10 aria-expanded:scale-[0.97] aria-expanded:opacity-70 subpixel-antialiased border-none bg-bone"
                  data-slot="trigger"
                  data-testid="actions-btn"
                  id="react-aria-:r2:"
                  type="button"
                >
                  <svg
                    class="text-blue-800/30 dark:text-white/30"
                    fill="currentColor"
                    height="14"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="14"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M328 256c0 39.8-32.2 72-72 72s-72-32.2-72-72 32.2-72 72-72 72 32.2 72 72zm104-72c-39.8 0-72 32.2-72 72s32.2 72 72 72 72-32.2 72-72-32.2-72-72-72zm-352 0c-39.8 0-72 32.2-72 72s32.2 72 72 72 72-32.2 72-72-32.2-72-72-72z"
                    />
                  </svg>
                </button>
              </td>
            </tr>
            <tr
              aria-labelledby="react-aria-:r0:-2-Name0"
              class="group outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 bg-white dark:bg-gray-400"
              data-id="2"
              data-key="2"
              data-last="true"
              data-odd="true"
              role="row"
              tabindex="-1"
            >
              <td
                class="relative align-middle whitespace-normal font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start first:rounded-l-lg last:rounded-r-lg text-[10px] sm:text-[14px] leading-[18px] py-[17px] px-[17px]"
                data-key="table-cell-0"
                id="react-aria-:r0:-2-Name0"
                role="rowheader"
                tabindex="-1"
              >
                <div
                  class="flex gap-3.5 items-center"
                >
                  <div
                    class="relative w-9 h-9 rounded-full"
                  >
                    <img
                      alt="customer avatar"
                      class="rounded-full "
                      data-nimg="fill"
                      decoding="async"
                      loading="lazy"
                      sizes="100vw"
                      src="http://localhost/_next/image?url=https%3A%2F%2Fimg-cdn.pixlr.com%2Fimage-generator%2Fhistory%2F65bb506dcb310754719cf81f%2Fede935de-1138-4f66-8ed7-44bd16efc709%2Fmedium.webp&w=3840&q=75"
                      srcset="/_next/image?url=https%3A%2F%2Fimg-cdn.pixlr.com%2Fimage-generator%2Fhistory%2F65bb506dcb310754719cf81f%2Fede935de-1138-4f66-8ed7-44bd16efc709%2Fmedium.webp&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fimg-cdn.pixlr.com%2Fimage-generator%2Fhistory%2F65bb506dcb310754719cf81f%2Fede935de-1138-4f66-8ed7-44bd16efc709%2Fmedium.webp&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fimg-cdn.pixlr.com%2Fimage-generator%2Fhistory%2F65bb506dcb310754719cf81f%2Fede935de-1138-4f66-8ed7-44bd16efc709%2Fmedium.webp&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fimg-cdn.pixlr.com%2Fimage-generator%2Fhistory%2F65bb506dcb310754719cf81f%2Fede935de-1138-4f66-8ed7-44bd16efc709%2Fmedium.webp&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fimg-cdn.pixlr.com%2Fimage-generator%2Fhistory%2F65bb506dcb310754719cf81f%2Fede935de-1138-4f66-8ed7-44bd16efc709%2Fmedium.webp&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fimg-cdn.pixlr.com%2Fimage-generator%2Fhistory%2F65bb506dcb310754719cf81f%2Fede935de-1138-4f66-8ed7-44bd16efc709%2Fmedium.webp&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fimg-cdn.pixlr.com%2Fimage-generator%2Fhistory%2F65bb506dcb310754719cf81f%2Fede935de-1138-4f66-8ed7-44bd16efc709%2Fmedium.webp&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fimg-cdn.pixlr.com%2Fimage-generator%2Fhistory%2F65bb506dcb310754719cf81f%2Fede935de-1138-4f66-8ed7-44bd16efc709%2Fmedium.webp&w=3840&q=75 3840w"
                      style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
                    />
                  </div>
                  <p
                    class="font-dm-sans text-blue-800 dark:text-white text-md text-nowrap"
                  >
                    Jane Smith
                  </p>
                </div>
              </td>
              <td
                class="relative align-middle whitespace-normal font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start first:rounded-l-lg last:rounded-r-lg text-[10px] sm:text-[14px] leading-[18px] py-[17px] px-[17px]"
                data-key="table-cell-1"
                role="gridcell"
                tabindex="-1"
              >
                <div
                  class="flex gap-2.5 items-center"
                >
                  <p
                    class="font-dm-sans text-blue-800 dark:text-white text-md text-nowrap"
                  >
                    jane.smith@example.com
                  </p>
                </div>
              </td>
              <td
                class="relative align-middle whitespace-normal font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start first:rounded-l-lg last:rounded-r-lg text-[10px] sm:text-[14px] leading-[18px] py-[17px] px-[17px]"
                data-key="table-cell-2"
                role="gridcell"
                tabindex="-1"
              >
                <p
                  class="font-dm-sans text-blue-800 dark:text-white text-md text-nowrap"
                >
                  987-654-3210
                </p>
              </td>
              <td
                class="relative align-middle whitespace-normal font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start first:rounded-l-lg last:rounded-r-lg text-[10px] sm:text-[14px] leading-[18px] py-[17px] px-[17px]"
                data-key="table-cell-3"
                role="gridcell"
                tabindex="-1"
              >
                <div
                  class="relative max-w-fit min-w-min inline-flex items-center justify-between box-border whitespace-nowrap h-7 text-small text-default-foreground w-[100px] px-2 rounded-[20px] bg-pink-500/10 dark:bg-gray-900"
                >
                  <span
                    class="flex-1 font-normal px-2 text-center text-md capitalize text-pink-500"
                  >
                    female
                  </span>
                </div>
              </td>
              <td
                class="relative align-middle whitespace-normal font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start first:rounded-l-lg last:rounded-r-lg text-[10px] sm:text-[14px] leading-[18px] py-[17px] px-[17px]"
                data-key="table-cell-4"
                role="gridcell"
                tabindex="-1"
              >
                <button
                  aria-expanded="false"
                  aria-haspopup="true"
                  class="group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none text-default-foreground data-[hover=true]:opacity-hover z-10 aria-expanded:scale-[0.97] aria-expanded:opacity-70 subpixel-antialiased border-none bg-bone"
                  data-slot="trigger"
                  data-testid="actions-btn"
                  id="react-aria-:r8:"
                  type="button"
                >
                  <svg
                    class="text-blue-800/30 dark:text-white/30"
                    fill="currentColor"
                    height="14"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="14"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M328 256c0 39.8-32.2 72-72 72s-72-32.2-72-72 32.2-72 72-72 72 32.2 72 72zm104-72c-39.8 0-72 32.2-72 72s32.2 72 72 72 72-32.2 72-72-32.2-72-72-72zm-352 0c-39.8 0-72 32.2-72 72s32.2 72 72 72 72-32.2 72-72-32.2-72-72-72z"
                    />
                  </svg>
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
